src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java

```diff
@@ -119,12 +116,17 @@ private OpenLoadFlowParameters getParametersExt(LoadFlowParameters parameters) {
             LOGGER.info("Slack bus selector: {}", slackBusSelector.getClass().getSimpleName());
             LOGGER.info("Voltage level initializer: {}", voltageInitializer.getClass().getSimpleName());
             LOGGER.info("Distributed slack: {}", parametersExt.isDistributedSlack());
+            LOGGER.info("Balance type: {}", parametersExt.getBalanceType());
             LOGGER.info("Reactive limits: {}", !parameters.isNoGeneratorReactiveLimits());
             LOGGER.info("Voltage remote control: {}", parametersExt.hasVoltageRemoteControl());
 
             List<OuterLoop> outerLoops = new ArrayList<>();
             if (parametersExt.isDistributedSlack()) {
-                outerLoops.add(new DistributedSlackOuterLoop());
+                if (parametersExt.getBalanceType() == OpenLoadFlowParameters.BalanceType.PROPORTIONAL_TO_GENERATION_P) {
+                    outerLoops.add(new DistributedGenerationSlackOuterLoop());
+                } else if (parametersExt.getBalanceType() == OpenLoadFlowParameters.BalanceType.PROPORTIONAL_TO_LOAD) {
+                    outerLoops.add(new DistributedLoadSlackOuterLoop());
+                }
```

Comment:
Throws an exception in case of not yet supported mode.

Reply:
Done.